airflow/models/taskinstance.py

```diff
@@ -547,7 +547,7 @@ def try_number(self):
         database, in all other cases this will be incremented.
         """
         # This is designed so that task logs end up in the right file.
-        if self.state in State.running:
+        if self.state == State.RUNNING:
```

Comment:
@ashb i think this resolves the "log to wrong file" problem.

when state is deferred, try number is decremented. so we need the plus one.